examples/08.memory_safety/memory_safety_inner.cc

```diff
@@ -140,9 +140,10 @@ int memory_safety_inner_entry(MemorySafetyBugClass operation)
 			volatilePointer = buf;
 			Capability tmp  = volatilePointer;
 			Debug::log("tmp: {}", tmp);
-			Debug::Assert(!tmp.is_valid(),
-			              "Code after storing stack pointer into global should "
-			              "be unreachable");
+			Debug::Assert(
+			  !tmp.is_valid(),
+			  "Code after accessing a stack pointer stored into global should "
```

Comment:
This message still needs fixing....

Reply:
Ooops.

Reply:
Doesn't my updated version capture what its now testing ?

Reply:
No, it isn't unreachable, which is why I added the dereference of the untagged pointer afterwards.  It should give back an untagged value.

Reply:
Ah, I was thinking that accessing tmp in the Debug at line 142 would trigger a tag violation, but of course you have to actually have to deference it  to get that